starknet_py/tests/e2e/conftest.py

```diff
@@ -84,3 +112,55 @@ def fixture_run_prepared_devnet(run_devnet) -> Tuple[str, dict]:
     net = run_devnet
     block = prepare_devnet(net)
     yield net, block
+
+
+@pytest.fixture(name="gateway_client", scope="function")
+def create_gateway_client(pytestconfig, run_devnet):
+    net = pytestconfig.getoption("--net")
+    net_address = {
+        "devnet": run_devnet,
+        "testnet": "testnet",
+        "integration": "https://external.integration.starknet.io",
+    }
+
+    return GatewayClient(net=net_address[net], chain=StarknetChainId.TESTNET)
+
+
+@pytest.fixture(name="rpc_client", scope="function")
+def create_rpc_client(run_devnet):
+    return FullNodeClient(
+        node_url=run_devnet + "/rpc", chain=StarknetChainId.TESTNET, net=run_devnet
+    )
+
+
+@pytest.fixture(name="account_client", scope="function")
+# pylint: disable=redefined-outer-name
+def create_account_client(pytestconfig, gateway_client):
+    net = pytestconfig.getoption("--net")
```

Comment:
I'd consider returning inside ifs instead of overwriting `client` and returning it at the end of the function